### ISSUE TITLE
AWS provider fails validation for the new  `il-central-1` AWS region

### ISSUE THREAD

**Issue by adrianrgould** [external-secrets/external-secrets#2586]
**Describe the bug**

Cluster secret stores using AWS provider fail validation when the region is `il-central-1`:

```
admission webhook "validate.clustersecretstore.external-secrets.io" denied the request: region not found: il-central-1
```

**To Reproduce**

Create a new  AWS SSM Parameter Store  backed  secret store with `region: il-central-1` :
```
---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    # ARN of IAM role to use when querying SSM
    eks.amazonaws.com/role-arn: arn:aws:iam::12346789023:role/my-iam-role
  name: my-service-account
  namespace: external-secrets
---
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  annotations:
  name: my-secret-store
spec:
  provider:
    aws:
      auth:
        jwt:
          serviceAccountRef:
            name: my-service-account
            namespace: external-secrets
      region: il-central-1
      service: ParameterStore
```

Kubernetes 1.27
ESO 0.9.2

**Expected behavior**
AWS region id `il-central-1` should pass validation

**Additional context**
This is the (very) new AWS region in Tel Aviv.  It opened for business on 8/1/2023.  I suspect an update to the AWS SDK client would resolve the issue.

**Comment by shuheiktgw** [comment]
Hi @adrianrgould 👋 I briefly looked into this issue, and below is where the validation runs. 

https://github.com/external-secrets/external-secrets/blob/5384954f46513a811425ce22c9186b2eb915f32e/pkg/provider/aws/provider.go#L95-L110

It seems https://github.com/external-secrets/external-secrets/pull/2568 updated the AWS SDK to v1.44.314, which includes the il-central-1 region. The change has been released in [the v0.9.2 release](https://github.com/external-secrets/external-secrets/releases/tag/v0.9.2), so IIUC if you use the latest version, it should work. I checked it locally just in case and confirmed the issue you described above happened in v0.9.1 but could not reproduce it with v0.9.2. Would you make sure that you really use the latest v0.9.2 ESO?

**Comment by adrianrgould** [comment]
Hi @shuheiktgw, thank you for looking into it.  You're correct, I was using v0.9.0, and not v0.9.2.   I had erroneously committed a kustomize chart cache which caused the version mismatch.  Gitignore updated!   I confirmed that il-central-1 is validating with v0.9.2.  Thanks again!
